adafruit_requests.py

```diff
@@ -37,35 +37,117 @@
 __repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_Requests.git"
 
 import errno
+import sys
+
+if sys.implementation.name == "circuitpython":
```

Comment:
Previously this pattern was used to only run the typing-related code in Python implementations which support it (such as CPython):

```python
try:
    # typing-related code here
    ...
except ImportError:
    pass
```

But that caused the CPython import failure issue described in #93 to be swallowed, so to help make debugging similar issues easier in the future, I replaced that pattern with this approach of reading `sys.implementation.name` to check if CircuitPython is being used or not instead.